File: solarbackend/src/main/java/solar/solarbackend/entity/Customer.java
package solar.solarbackend.entity;

import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import com.fasterxml.jackson.annotation.JsonIgnore; // Import this annotation
import lombok.Data;
import java.util.List;

@Entity
@Data
@Table(name = "Customer")
public class Customer {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "cus_id")
    private Long id;

    @Column(name = "name", nullable = false)
    private String name;

    @Column(name = "address")
    private String address;

    @Column(name = "number")
    private String number;

    @Column(name = "e_mail")
    private String email;

    @Column(name = "nic", unique = true)
    private String nic;

    @JsonIgnore
    @OneToMany(mappedBy = "customer", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    private List<Project> projects;
    @JsonIgnore
    @OneToMany(mappedBy = "customer", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    private List<BankAccount> bankAccounts;

}

File: frontend/src/app/customer.ts
import { Project } from './project';
import { BankAccount } from './bank-account';
export class Customer {
    id: number;
    name: string;
    address: string;
    number: string;
    email: string;
    nic: string;
    projects: Project[]; // Assuming you have a 'Project' class in Angular
    bankAccounts: BankAccount[]; // Assuming you have a 'BankAccount' class in Angular
  
    constructor(
      id: number = 0,
      name: string = '',
      address: string = '',
      number: string = '',
      email: string = '',
      nic: string = '',
      projects: Project[] = [],
      bankAccounts: BankAccount[] = []
    ) {
      this.id = id;
      this.name = name;
      this.address = address;
      this.number = number;
      this.email = email;
      this.nic = nic;
      this.projects = projects;
      this.bankAccounts = bankAccounts;
    }
  }
  